packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter.tsx

```diff
@@ -81,17 +83,42 @@ function formatFacetValuesForDisplay(
   }
 }
 
+/**
+ * Based on the uncommitted query, should the "All" checkbox be selected?
+ * This only considers filters applied via selectedFacets. Filters applied via
+ * sql or additionalFilters will be ignored.
+ *
+ * @returns true iff no values have been selected for this facet filter
+ * @param nextQueryRequest
+ * @param columnModel
+ */
+function getAllIsSelected(
+  nextQueryRequest: ReadonlyDeep<QueryBundleRequest>,
+  columnModel: SelectColumn,
+) {
+  return !nextQueryRequest.query.selectedFacets?.find(
+    facetRequest =>
+      facetRequest.concreteType ===
+        'org.sagebionetworks.repo.model.table.FacetColumnValuesRequest' &&
+      facetRequest.columnName === columnModel.name,
+  )
+}
```

Comment:
Extracted from component. Logic was changed to use the _request_ object rather than the _response_ object, so it should more accurately track with what the user is about to do, instead of what the system most recently did.

Reply:
I would think that this logic would fix https://sagebionetworks.jira.com/browse/PORTALS-2680.  Is this accurate?

Reply:
I think it will! I'll add the issue to my sprint and double-check to confirm